### ISSUE TITLE
ng-class-even does not work for negative $index values

### ISSUE THREAD

**Issue by mfeingold** [angular/angular.js#5204]
Apparently ng-class-even thinks that all negative values are odd. Interestingly enough, ng-class-odd works properly for both positive and negative values, which is inconsistent with the common definition of 'even'

Of course it is a moot point when using ng-class-even with the ng-repeat because ng-repeat never generates negative values, but it does become an issue with the [ng-scroll] (https://github.com/Hill30/NGScroller) directive.

In other words I am not sure if this issue should be considered to be a bug or a feature request.

The issue can be easily fixed and I would gladly do it if you would accept it.

And here is the [repro](http://rawgithub.com/Hill30/NGScroller/master/src/windowviewport.html)

**Comment by lgalfaso** [comment]
This is already fixed, so the issue can be closed
